### DISCUSSION TITLE
Utility for getting serialized types of loader

### DISCUSSION THREAD

**Discussion by JeffersonBledsoe** [remix-run/remix#3818] [Q&A] [answered]
As noted in [Decision 0003](https://github.com/remix-run/remix/blob/main/decisions/0003-infer-types-for-useloaderdata-and-useactiondata-from-loader-and-action-via-generics.md), `useLoaderData` returns the serialized version of the data, so types such as `Date` become `string`. Is there a recommended way to conveniently get the correct type of the serialized version of the data?

**Comment by kiliman** [comment, marked as answer]
Checkout https://rmx.fyi/superjson

I have a wrapper that uses `superjson` to get full-fidelity types from loader to useLoaderDate, with type inference.

**Reply by JeffersonBledsoe** [reply]
Hey, thanks for the link, this looks great! I'll give it a try later. Would still be interested to know if it's doable with 'vanilla' remix

**Reply by kiliman** [reply]
Not at the moment. Data returned from loader/action is serialized to JSON, so non-JSON types will be converted to strings. It's your responsibility to convert them back, hence my helper function. `superjson` adds 9KB to production builds, so they are hesitant to add to core Remix. I'm writing a smaller/faster replacement to `superjson`, but it will still probably be an optional import.
